pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -72,7 +72,7 @@ func NewClientAccessFactory(optionalClientConfig clientcmd.ClientConfig) ClientA
 
 // NewClientAccessFactoryFromDiscovery allows an external caller to substitute a different discoveryFactory
 // Which allows for the client cache to be built in ring0, but still rely on a custom discovery client
-func NewClientAccessFactoryFromDiscovery(flags *pflag.FlagSet, clientConfig clientcmd.ClientConfig, discoveryFactory DiscoveryClientFactory) ClientAccessFactory {
+func NewClientAccessFactoryFromDiscovery(flags *pflag.FlagSet, clientConfig clientcmd.ClientConfig, discoveryFactory *discoveryFactory) ClientAccessFactory {
```

Comment:
changing this to a concrete, unexported type breaks the ability to inject a different discovery factory. consider adding `BindFlags(flags *pflag.FlagSet)` to the `DiscoveryClientFactory` interface instead

cc @deads2k